docs/blog/2019-02-06-government-open-data-site-with-gatsby/index.md

```diff
@@ -0,0 +1,237 @@
+
+### Modularity
+
+Perhaps React's rising popularity is due to its modularity in the form of React components. Our site features repeating instances of similar or identical patterns, so Gatsby's use of React was attractive to our team.
+
+For example, the homepage's `KeyStatsSection` component references the `StackedBarChartLayout` component, passing it props:
+
+```jsx
+return (
+  <div is="chart">
+    <StackedBarChartLayout
+      dataSet={this.state[dataSetId]}
+      title={title}
+      styleMap={CHART_STYLE_MAP}
+      sortOrder={CHART_SORT_ORDER}
+      legendTitle={CHART_LEGEND_TITLE}
+      legendDataFormatFunc={dataFormatFunc || utils.formatToCommaInt}
+      barSelectedCallback={this.dataKeySelectedHandler.bind(
+        this,
+        dataSetId,
+        this.state[dataSetId].syncId
+      )}
+    />
+  </div>
+)
+```
+
+#### Pattern library
+
+This component architecture also feeds our pattern library (not yet live), so when we build or change a component, our pattern library will update itself. We will then have a working inventory of our components to reference for new feature development.
+
+#### CSS Modules
+
+With our homepage rebuild, we also started using [CSS Modules](https://github.com/css-modules/css-modules). CSS Modules are CSS files "in which all class names and animation names are scoped locally by default," and they're a great way to avoid selector name collisions and limit the accumulation of unused styles.
+
+Scoping and storing each component's CSS locally makes it much easier to track down and edit a component's styling.
+
+### A modern framework
+
+Nothing is future-proof in technology, but migrating to Gatsby gets us closer, with a flexible, modular architecture and a growing and spirited community of developers behind it. By migrating to Gatsby, we're trying to build a maintainable, future-friendly codebase for the team that comes after us.
+
+## Challenges during rebuild
+
+While our experience refactoring the homepage was largely positive, we have faced some challenges migrating to Gatsby. Some of those challenges are inherent in refactoring, caused by maintaining two codebases and build processes simultaneously. Others are perhaps due to the relatively new emergence of Gatsby itself, which is actively being worked on and upgraded.
```

Comment:
Gatsby isn't really _all that new_! We'd love any feedback on areas we can improve upon, here re: challenges you faced.

Reply:
We are experiencing relatively long build times, @DSchau. Part of the reason is that we're still compiling two separate codebases, but the build time increased a lot when we expanded our graphql schema. We hope eliminating the Jekyll build, and upgrading our Gatsby version, will help mitigate this problem, but we're more than happy to hear tips from Gatsby folks on how we might improve the build.